libbeat/autodiscover/autodiscover.go

```diff
@@ -105,11 +105,12 @@ func (a *Autodiscover) Start() {
 	logp.Info("Starting autodiscover manager")
 	a.listener = a.bus.Subscribe(a.adapter.EventFilter()...)
 
+	// Start the worker before starting providers as the bus can be bounded
+	go a.worker()
```

Comment:
Let's add a more detailed comment explaining that worker reads events generated by providers. Moving `go a.worker()` after the loop leads to a deadlock in the providers, due to a missing consumer on the channels. It's too easy to not see this kind of issue and accidentily break the code again.